src/index.ml

```diff
@@ -323,6 +403,7 @@ struct
       with Not_found ->
         let instance = v ~fresh ~readonly ~log_size root in
         Hashtbl.add roots (root, readonly) instance;
+        if readonly then sync_log instance;
```

Comment:
Bit difficult to see what's going on here, but doesn't this try to sync the log after creating a fresh instance on disk? (In at least two code paths, but maybe not all.)

Reply:
It can be that the rw instance is already open. When the ro instance is opened, it does not automatically sync its log with the one on disk (see https://github.com/mirage/index/blob/master/src/index.ml#L340). 
 
I added a [test](https://github.com/mirage/index/pull/175/files#diff-22a049f4c6a8354008c166e8f65b02b6R268) that fails without this.